### PR TITLE
Data saving and reading

### PR DESCRIPTION
**Pull request by VytalyGorbatov**
@Crowy-o developed subsystem that handles saving data in files according to it's creation time;
@VytalyGorbatov developed subsystem that handles finding data according to client's request;

### CODE REVIEW

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	bool success = true;
+	CLOG_DEBUG_START_FUNCTION( );
+	success = m_json_formatter.TryEraseAllData( );
+	CLOG_DEBUG_END_FUNCTION_WITH_RETURN(success);
+	return success;
+}
+
+bool CDrivesInfoJSONDatabase::InsertCommitedData( )
+{
+	bool success = true;
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:49]: By default function should return false

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	CLOG_TRACE_VAR_CREATION(end);
+
+	auto json = m_json_formatter.GetJSONFormattedData( );
+	CLOG_TRACE_VAR_CREATION(json);
+	auto date_iter = json->back().find("date");
+	CLOG_TRACE_VAR_CREATION(date_iter);
+
+	do
+	{
+		if (date_iter == json->back( ).end( ))
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:63]: Invert conditionals and remove success = false under each conditional

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+			success = false;
+			break;
+		}
+		if (!m_reader.InsertData(begin, end, date))
+		{
+			CLOG_ERROR("Cant insert data into database");
+			success = false;
+			break;
+		}
+	} while (false);
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:106]: add success = true before } while (false);

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+#include "Utils.h"
+#include "Log.h"
+
+#include "CDrivesInfoJSONDatabase.h"
+
+CDrivesInfoJSONDatabase::CDrivesInfoJSONDatabase(
+	const std::filesystem::path& path_to_file)
+	: CMonitoringInfoJSONDatabase(path_to_file)
+{
+	CLOG_DEBUG_START_FUNCTION( );
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:15]: We do not need such logs in Ctors

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	CLOG_DEBUG_END_FUNCTION_WITH_RETURN(success);
+	return success;
+}
+
+bool CDrivesInfoJSONDatabase::InsertCommitedData( )
+{
+	bool success = true;
+	CLOG_DEBUG_START_FUNCTION( );
+	std::streampos begin;
+	CLOG_TRACE_VAR_CREATION(begin);
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:52]: What does this macros do CLOG_TRACE_VAR_CREATION(begin);? not Sure if he needed

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+	auto json = m_json_formatter.GetJSONFormattedData( );
+	CLOG_TRACE_VAR_CREATION(json);
+	auto date_iter = json->back().find("date");
+	CLOG_TRACE_VAR_CREATION(date_iter);
+
+	do
+	{
+		if (date_iter == json->back( ).end( ))
+		{
+			CLOG_ERROR("Cant find date in json block");
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:65]: "Cant find date in json block" -> "Failed to find date in json block"

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+			success = false;
+			break;
+		}
+		if (!date_iter.value( ).is_string( ))
+		{
+			CLOG_ERROR("Cant find date in json block");
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:71]: The same error logged as below, error messages should be unique

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+			success = false;
+			break;
+		}
+		std::string date_str = *date_iter;
+		time_t date;
+		if (!Utils::StringToDate(date_str, m_date_format, date))
+		{
+			CLOG_ERROR("Cant convert string to time_t");
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:79]: Just failed to convert date

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+			success = false;
+			break;
+		}
+
+		std::lock_guard<std::mutex> guard(m_lock);
+		std::string new_path = m_path_to_file.string( );
+		if (!m_p_path_constructor->UpdatePathToFile(new_path))
+		{
+			CLOG_ERROR("Cant update path to file");
```
Comment (AntonVoytov) [inline Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:88]: Failed to update path '%s' , new_path.c_str()

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 3 above)
Comment (AntonVoytov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:106]: Please take a look at all logs and rephrase them if needed

**Review by AntonVoytov — COMMENTED** [review]
Review the comments fix them if possible

File: utils/include/Utils.cpp [hunk 10]
```diff
@@ -238,3 +239,72 @@ EMemoryConvertType Utils::DefineCountType(int count_type_from_xml)
...
+        return true;
+    }
+
+    return std::filesystem::is_directory(path);
+}
+
+bool Utils::IsDayPassed(std::string& day)
+{
+    std::string curr_day;
+    if (Utils::TryGetCurrentDateAndTimeFormatted(curr_day, "%d"))
```
Comment (AntonVoytov) [inline utils/include/Utils.cpp:276]: Is Utils::TryGetCurrentDateAndTimeFormatted the same as TryGetCurrentDateAndTimeFormatted?

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 1 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Fixed

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 2 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Fixed

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 3 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp:94]: Both fixed

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 4 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Deleted

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 5 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Deleted

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 6 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Fixed

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 7 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Changed

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 8 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Changed

File: Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp (hunk 9 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CDrivesInfoJSONDatabase.cpp]: Changed

File: utils/include/Utils.cpp (hunk 10 above)
Comment (VytalyGorbatov) [reply utils/include/Utils.cpp:276]: Utils:: deleted

**Pull request merged**